pkg/controller/middleware/firewall.go

```diff
@@ -77,7 +77,17 @@ func ProcessFirewall(h render.Renderer, typ string) mux.MiddlewareFunc {
 				ipStr = strings.Split(xff, ",")[0]
 			}
 
+			// In some cases, the remote addr will include a port. However, Go doesn't
+			// make it easy to distinguish between an ip:port and an IPv6 address.
+			// Here we'll attempt to split the address into host:port, but if that
+			// fails, we'll attempt to process the original value as an IP directly.
+			host, _, err := net.SplitHostPort(ipStr)
```

Comment:
Unrelated to this PR, but now I want to use `net.SplitHostPort` for:
https://github.com/google/exposure-notifications-verification-server/blob/c5acc33cfcd11fcb3ec1bc26f0660622c606011c/pkg/controller/associated/index.go#L48

https://github.com/google/exposure-notifications-verification-server/blob/c5acc33cfcd11fcb3ec1bc26f0660622c606011c/pkg/controller/redirect/index.go#L35

Reply:
https://github.com/google/exposure-notifications-verification-server/pull/1454